engine/lite/request-response/runtime/src/main/scala/pl/touk/nussknacker/engine/requestresponse/openapi/RequestResponseOpenApiGenerator.scala

```diff
@@ -1,47 +1,94 @@
 package pl.touk.nussknacker.engine.requestresponse.openapi
 
+import akka.http.scaladsl.model.{HttpHeader, HttpRequest}
+import com.typesafe.scalalogging.LazyLogging
 import io.circe.Json
 import io.circe.syntax._
-import pl.touk.nussknacker.engine.requestresponse.RequestResponseInterpreter.RequestResponseScenarioInterpreter
 import pl.touk.nussknacker.engine.requestresponse.api.openapi.RequestResponseOpenApiSettings.OPEN_API_VERSION
 import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
 
-import scala.language.higherKinds
+import scala.collection.JavaConverters._
 
-object RequestResponseOpenApiGenerator {
+class RequestResponseOpenApiGenerator(oApiInfo: OApiInfo, serversDescription: Option[OApiServer], additionalInvocationPaths: List[String],
+                                      invocationPathFromDefinitionPath: List[String] => List[String]) extends LazyLogging {
```

Comment:
why is it defined in ScenarioRoute? When would we want to provide different function?
Also, shouldn't we have same logic in line 24?

Reply:
Ad1. Because this "one level up" logic is related with how we organize routes in ScenarioRoute.
Ad2. invocationPathFromDefinitionPath is invoked on both originalPathSegments and request.getUri() segments - there is getOrElse before invocation of this method